### DISCUSSION TITLE
EEP G5-3F-7F and A5-3F-7F

### DISCUSSION THREAD

**Discussion by SteffenS27** [mak-gitdev/HA_enoceanmqtt#99] [Request]
Hi mak-dev,
My problem now is to integrate the FSB covers with their status. The problem is, that I found that there is a EEP G5-3F-7F and A5-3F-7F used to transmit the status. But it is not so easy for me to implement that in the EEP.xml - could you make any suggestion to that - found out some others would be also interested for that.
Thanks in advance.

**Comment by mak-gitdev** [comment]
Hi @SteffenS27,

This is related to the general support of Eltako devices.
I am working on it and I may release support in 2-3 weeks.

**More details**
The real issue is not on the support of these EEP in the EEP file (A5-3F-7F is now available in the latest version) but in the fact that the current version supports only one EEP per address.
FSB covers require 2 EEP per address, one to send open/closed status similar to F6-02-01 and a more detailed one using A5-3F-7F to indicate the running time when the device is stopped while the cover is not fully open/closed.

With the modifications I am making for Eltako users, you will only have to add in the device configuration:
```
[FSB61NP]
address = 0xDEADBEEF
model    = eltako/FSB61NP (syntax may change)
```
And you will have in HA something like:
![image](https://github.com/mak-gitdev/HA_enoceanmqtt/assets/73297795/eb0239b8-27a6-44c6-8575-92cab6493dc2)
- One button to send the pairing telegram to pair your EnOcean transceiver to your Eltako device.
- The cover entity.
- A number entry to configure the time needed to close/open the cover as it is needed to calculate the position.

The same philosophy will apply to all Eltako devices and as you can see, there will be no more need to use virtual switches (this was a temporary solution).

**Reply by cubertt** [reply]
That will be great! Happy to hear that @mak-gitdev

**Reply by cubertt** [reply]
I will try it here: i would like to help test the preview version, as i wrote in the homeassistant forum!

**Comment by UsefulVid** [comment]
I have an FSB12 and its controlling 2 motors.
So I need 4 Buttons :)
```
[FSB61NP]
address = 0xDEADBEEF
model    = eltako/FSB61NP (syntax may change)
```

is this already usable or not published? If you need somebody to test your changes, please let me know.
